SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -366,27 +369,47 @@ - (void)didEnterStateRegistered {
-    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
-    SDLLanguage actualLanguage = self.registerResponse.language;
-    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:)];
-    
+    // the language represents hmi language
+    SDLLanguage desiredHMILanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage desiredVRLanguage = self.currentVRLanguage;;
+
+    SDLLanguage actualHMILanguage = self.registerResponse.hmiDisplayLanguage;
+    SDLLanguage actualVRLanguage = self.registerResponse.language;
+
+    BOOL oldDelegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:)];
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:hmiLanguage:)];
+
     // language mismatch? but actual language is a supported language? and delegate has implemented method?
-    if (![actualLanguage isEqualToEnum:desiredLanguage] && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
-        [self sdl_transitionToState:SDLLifecycleStateUpdatingConfiguration];
+    if ((delegateCanUpdateLifecycle || oldDelegateCanUpdateLifecycle)
+        && ([supportedLanguages containsObject:actualHMILanguage] || [supportedLanguages containsObject:actualVRLanguage])) {
+        if (![actualHMILanguage isEqualToEnum:desiredHMILanguage]
+            || ![actualVRLanguage isEqualToEnum:desiredVRLanguage]) {
+            [self sdl_transitionToState:SDLLifecycleStateUpdatingConfiguration];
+        } else {
+            [self sdl_transitionToState:SDLLifecycleStateSettingUpManagers];
+        }
     } else {
         [self sdl_transitionToState:SDLLifecycleStateSettingUpManagers];
     }
 }
 
 - (void)didEnterStateUpdatingConfiguration {
     // We can expect that the delegate has implemented the update method and the actual language is a supported language
+    SDLLanguage actualHMILanguage = self.registerResponse.hmiDisplayLanguage;
     SDLLanguage actualLanguage = self.registerResponse.language;
-    SDLLogD(@"Updating configuration due to language mismatch. New langugage: %@", actualLanguage);
+    SDLLogD(@"Updating configuration due to language mismatch. New language: %@, new hmiLanguage: %@", actualLanguage, actualHMILanguage);
 
-    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerShouldUpdateLifecycleToLanguage:actualLanguage];
+    SDLLifecycleConfigurationUpdate *configUpdate = nil;
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:hmiLanguage:)];
+    if (delegateCanUpdateLifecycle) {
+        configUpdate = [self.delegate managerShouldUpdateLifecycleToLanguage:actualLanguage hmiLanguage:actualHMILanguage];
+    } else {
+        configUpdate = [self.delegate managerShouldUpdateLifecycleToLanguage:actualLanguage];
```

Comment:
This creates a warning, you have to wrap the deprecation in a pragma to suppress the warning. This is done elsewhere in the project.